### PR TITLE
Refactor/big

### PR DESCRIPTION
**Pull request by polomani**
- split tests into separate files for each big chunk
- removed some useless tests
- added more unit tests
- separated logic that works with visual feedback from request logic (assigning labels, chunking etc)
- replaced some utility functions with npm modules
- removed `res.error = '...'` in favour of `throw new Error()`
- split logic into small pieces (still needs improvement)
- improved readability

### CODE REVIEW

File: src/cdfDatasource.ts [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+    responseTimeseries.push(...series);
+  });
+
+  return responseTimeseries;
+}
+
+export function datapoints2Tuples<T extends Timestamp[]>(
+  datapoints: T,
+  aggregates
```
Comment (mrtraser) [inline src/cdfDatasource.ts]:
```typescript
aggregates: string
```
?

File: src/cdfDatasource.ts [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+          }
+        }
+      }
+    }
+  }
+  return labels;
+}
+
+async function getTimeseriesLabel(
+  label,
```
Comment (mrtraser) [inline src/cdfDatasource.ts]:
```typescript
getTimeseriesLabel(
  label: string,
  externalId: string,
  target: QueryTarget,
  connector: Connector
): Promise<string> {...}
```

File: src/__tests__/datasource.test.ts [hunk 3]
```diff
@@ -0,0 +1,817 @@
...
+    });
+  });
+
+  describe('Given an older queryTarget format', () => {
+    let result;
+    const oldTarget: QueryTarget = {
+      aggregation: 'average',
+      refId: 'A',
+      target: 'Timeseries123',
+      granularity: undefined,
```
[the remark this answers is not in the document]
Comment (polomani) [reply src/__tests__/datasource.test.ts]: no, removed

**Comment by polomani** [comment]
I just merge it to v1 https://github.com/cognitedata/cognite-grafana-datasource/tree/feat/v1 branch, so we can do the main work based on it

**Pull request merged**